azure-core/azure/core/pipeline/policies/universal.py

```diff
@@ -0,0 +1,310 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+"""
+This module is the requests implementation of Pipeline ABC
+"""
+from __future__ import absolute_import  # we have a "requests" module that conflicts with "requests" on Py2.7
+import contextlib
+import json
+import logging
+import os
+import platform
+import threading
+from typing import TYPE_CHECKING, cast, List, Callable, Iterator, Any, Union, Dict, Optional  # pylint: disable=unused-import
+import xml.etree.ElementTree as ET
+import warnings
+import types
+import re
+
+from azure.core import __version__  as azcore_version
+from .base import HTTPPolicy, SansIOHTTPPolicy
+from urllib3 import Retry  # Needs requests 2.16 at least to be safe
+
+from azure.core.exceptions import (
+    DecodeError,
+    raise_with_traceback
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HeadersPolicy(SansIOHTTPPolicy):
+    """A simple policy that sends the given headers
+    with the request.
+
+    This overwrite any headers already defined in the request.
+    """
+    def __init__(self, headers=None, **kwargs):
+        # type: (Mapping[str, str]) -> None
+        self.headers = headers or {}
+
+    def on_request(self, request, **kwargs):
+        # type: (PipelineRequest, Any) -> None
+        additional_headers = kwargs.pop('headers', {})
```

Comment:
This implementation will replace a value, but it doesn't allow me to append an additional value for an arbitrary header. I *think* this semantic is reasonable. 

The two twists are usually that you can have multiple values for any given key and that one sometimes want to clear a header value (which, in turn, is accomplished by passing in a `None` for the value in most systems that I'm aware of).

Reply:
Do we have "standardized" behaviour in the guidelines for this kind of thing? Or is it simply a matter of us picking an approach and documenting it consistently?